SCons/Tool/MSCommon/vc.py

```diff
@@ -939,6 +1125,13 @@ def get_installed_vcs(env=None):
     if __INSTALLED_VCS_RUN is not None:
         return __INSTALLED_VCS_RUN
 
+    save_target_arch = env.get('TARGET_ARCH', UNDEFINED) if env else None
```

Comment:
Why are we saving and deleting `TARGET_ARCH` here?

Reply:
> Why are we saving and deleting TARGET_ARCH here?

The discussion for #4312 contained some of the details.  Repeated here for convenience.

https://github.com/SCons/scons/issues/4312#issuecomment-1452526557
> At first glance, I'm a little concerned about why it reports that no versions of the MSVC compiler were found:
> 
> ```
> scons: warning: unsupported host, target combination ('amd64', 'arm64') for MSVC version 11.0
> File "C:\Users\mats\github\scons\TRY.THIS.4", line 1, in <module>
> 
> scons: warning: MSVC version '11.0' was not found.
>   No versions of the MSVC compiler were found.
>   Visual Studio C/C++ compilers may not be set correctly
> ```

https://github.com/SCons/scons/issues/4312#issuecomment-1452771341
>... The root problem is with the construction of the installed VCS cache list (i.e., `__INSTALLED_VCS_RUN`) in vc.py.
> 
> The installed VCS list is cached (i.e., computed once). However, there are indirect dependencies on values from the environment. Most notably, the TARGET_ARCH.
> 
> In some of the tests, when the installed VCS cache list is constructed, the TARGET_ARCH (e.g., `arm` and `arm64`) does not exist in VS2022 and therefore there are no detected VCS.
> 
> The quick fix below uses an undefined TARGET_ARCH during the initial construction and then restores the original value prior to exit. This should be enough to accurately detect all installations with at least one usable HOST/TARGET combination.
>...